agent/src/softwarecontaineragent.cpp

```diff
@@ -119,94 +120,113 @@ bool SoftwareContainerAgent::triggerPreload()
     return true;
 }
 
-inline bool SoftwareContainerAgent::isIdValid(ContainerID containerID)
+void SoftwareContainerAgent::assertContainerExists(ContainerID containerID)
 {
-    return (containerID < INT32_MAX)
-        && (containerID >= 0)
-        && (1 == m_containers.count(containerID));
+    if (containerID >= INT32_MAX) {
+        std::string errorMessage("Invalid Container ID: "
+                                + std::to_string(containerID)
+                                + ". ID can not be greater than INT32");
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
```

Comment:
I think we should try to log as quickly as possible when something happens to lessen the time spent debugging. You could argue that this can be done as well by tracing the exception, but that is not how we handle exceptions AFAIK.